### ISSUE TITLE
ERROR:absl:dm_control viewer intercepted an environment error. Original message: 'tuple' object has no attribute 'last'

### ISSUE THREAD

**Issue by KJaebye** [deepmind/dm_control#364]
Hello, I am using dm_control.viewer to render a trained agent. My code is like this:
```
def visualize_agent(mean_action=True):
        env = self.env

        from dm_control import viewer

        def policy_fn(time_step):
            cur_state = torper.tensor([tools.get_state_flatten(time_step.observation)], device=self.device)
            action = self.policy_net.select_action(cur_state, mean_action).detach().numpy()
            return action

        viewer.launch(environment_loader=env, policy=policy_fn)
```
```
    """ create agent """
    agent = HopperAgent(cfg, logger, dtype=dtype, device=device, seed=cfg.seed, num_threads=1,
                        render=True, training=False, checkpoint=epoch)

    agent.visualize_agent()
```

I got an error like this: 

```
ERROR:absl:dm_control viewer intercepted an environment error.
Original message: 'tuple' object has no attribute 'last'
dm_control viewer intercepted an environment error.
Original message: 'tuple' object has no attribute 'last'
Traceback:
  File "/Users/kjaebye/opt/miniconda3/envs/mjctest/lib/python3.9/site-packages/dm_control/viewer/runtime.py", line 256, in _step
    finished = self._time_step.last()
```
I check the code in viewer. I guess the viewer.launch shall create a class Runtime, and run _step() . In _step():
```
  def _step(self):
    """Generates an action and applies it to the environment.

    If a `policy` was provided, this will be invoked to generate an action to
    feed to the environment, otherwise a default action will be generated.

    Returns:
      A boolean value, True if the environment signaled the episode end, False
      if the episode is still running.
    """
    finished = True
    with self._error_logger:
      if self._policy:
        action = self._policy(self._time_step)
      else:
        action = self._default_action
      self._time_step = self._env.step(action)
      self._last_action = action
      finished = self._time_step.last()
    return finished or self._error_logger.errors_found
```

However, I have indicated the policy policy_fn() so it cannot get into 'else' part, but error occur. What is the mechanism of this?

**Comment by KJaebye** [comment]
It seems like the last problem that I met [https://github.com/deepmind/dm_control/issues/355](url). The last error disapeared without any fix. Could someone get the same one and can explain this?

**Comment by KJaebye** [comment]
I found what's wrong with this. The reason is I wrote a new class function 'step' in object Environment where I return two values 'time_step' and 'done'. However, when using viewer.launch() and set a policy, mujoco might run the object env and execute step() but expect only one value time_step as return. Keeping the return value as original can resolve this.

However, I think this setting may limit the user's freedom to make changes at will. If I would like to change step() function, I cannot use dm_control.viewer to render.
